webrtc.html

```diff
@@ -9287,7 +9287,15 @@ <h4 id="sctp-transport-connected">Connected procedure</h4>
                   <p>Let <var>channel</var> be the <code><a>RTCDataChannel</a></code> object.</p>
                 </li>
                 <li>
-                  <p>If the <var>channel</var>'s <a>[[\DataChannelId]]</a> slot is greater or
+                  <p>If <var>channel</var>'s <a>[[\DataChannelId]]</a> slot is
+                    <code>null</code>, initialize <a>[[\DataChannelId]]</a>
+                    to a value generated by the
+                    user agent, according to [[!RTCWEB-DATA-PROTOCOL]].
```

Comment:
It's important that the underlying data transport's ID matches this label. I think we either need to say that this generation took place as part of the underlying connection procedure and is surfaced here, or we have to generate the ID here and then "in parallel" update the underlying data transport to match this ID.

Reply:
I took "generated by the user agent" to be so obvioiusly the underlying data transport's ID that it didn't need to be called out.

Reply:
Clarification:

Each RTCDataChannel has an associated underlying data transport.
Whenever something happens to the underlying transport, like connecting or closing which could happen in parallel on a network thread, we queue a task to update the RTCDataChannel to match. We need to be careful about keeping these two objects (the JS version and the actual SCTP implementation) in sync.

For example, when creating an RTCDataChannel with createDataChannel(), there's a step to "Create channel's associated underlying data transport and configure it according to the relevant properties of channel." meaning we pass down the ID to the underlying transport. Likewise when a data channel is created by the underlying sctp transport, we queue a task to to create an RTCDataChannel to match the configuration received from the underlying data transport.

So if we modify the [DataChannelId] internal slot, we need to inform the underlying data transport that this is the ID that it should use.

Do we need to add "In parallel, update the underlying data transport's id"?

Of can we say "If [DataChannelId] is null, update it to match the ID received from the underlying sctp data channel that was generated as specified in [RTCWEB-DATA-PROTOCOL]"?

Reply:
> I took "generated by the user agent" to be so obvioiusly the underlying data transport's ID that it didn't need to be called out.

The concern isn't so much that they mismatch since that clearly is the intent but that it is unclear if the ID is generated before or after the queuing of a task.

Reply:
The id is not a *modifiable* property. It's a *settable* property. Once it's set to a non-null value, it can't be changed. So we don't have to deal with synchronization of changes - either it's set at creation (in JS, communicated to lower layers) or it's set in the protocol machine (in lower layers, communicated to JS). This makes thinking about synchronization somewhat easier.

Reply:
I changed this to "initialize [[DataChannelId]] to the value generated by the underlying sctp data channel, according to..."